squid: qa: Add multifs root_squash testcase

kotreshhr opened the pull request:
backport tracker: https://tracker.ceph.com/issues/64939

---

backport of https://github.com/ceph/ceph/pull/55833
parent tracker: https://tracker.ceph.com/issues/64641

----

backport tracker: https://tracker.ceph.com/issues/65907

backport of https://github.com/ceph/ceph/pull/56846
parent tracker: https://tracker.ceph.com/issues/65246

Bot the trackers need to be backported as the https://github.com/ceph/ceph/pull/55833 has a bug.

rishabh-d-dave committed:
qa/cephfs: run test_multifs_single_path_rootsquash for kclient too

Root squash is valid for kclient too, Kotresh ran test recently fo it
against main branch. Therefore it is safe to remove.
https://github.com/ceph/ceph/pull/56846#discussion_r1587507868

Signed-off-by: Rishabh Dave <ridave@redhat.com>
(cherry picked from commit 422d98c01043e6f0b99730ffd5b50eabe230aec9)

rishabh-d-dave committed:
qa/cephfs: fix test_multifs_single_path_rootsquash

test_multifs_single_path_rootsquash was never run with vstart_runner.py
or with teuthology and is therefore full of bugs. Fix it to make sure it
runs fine.

Introduced-by: 1fda8ed2d4a9
Fixes: https://tracker.ceph.com/issues/65246
Signed-off-by: Rishabh Dave <ridave@redhat.com>
(cherry picked from commit d9752a69735e64e826f5ac688def375cfac1077b)

rishabh-d-dave commented:
Marking this as `DNM` because this test is broken, which wasn't discovered since it wasn't tested with vstart_runner.py or teuthology. A fix has been written for it (https://github.com/ceph/ceph/pull/56846). Both of them needs to be backported together to avoid passing same bug to release branches.

joscollin commented:
jenkins test make check

kotreshhr commented:
> Marking this as `DNM` because this test is broken, which wasn't discovered since it wasn't tested with vstart_runner.py or teuthology. A fix has been written for it (#56846). Both of them needs to be backported together to avoid passing same bug to release branches.

@rishabh-d-dave Would you take care of this ? or should I backport #56846 with this ?

rishabh-d-dave commented:
> > Marking this as `DNM` because this test is broken, which wasn't discovered since it wasn't tested with vstart_runner.py or teuthology. A fix has been written for it (#56846). Both of them needs to be backported together to avoid passing same bug to release branches.
> 
> @rishabh-d-dave Would you take care of this ? or should I backport #56846 with this ?

IIRC, last time checked despite of #56846 this test is not doing fine, I'll re-check.

joscollin commented:
@kotreshhr Please fix the conflicts

kotreshhr committed:
qa: Add multifs root_squash testcase

Fixes: https://tracker.ceph.com/issues/64641
Signed-off-by: Kotresh HR <khiremat@redhat.com>
(cherry picked from commit 1fda8ed2d4a97b5d8e882ada2dd4b582c4da622b)

Conflicts:
  qa/tasks/cephfs/test_admin.py - Few of the other tests got added
in between.

kotreshhr commented:
> @kotreshhr Please fix the conflicts

Done

kotreshhr commented:
> Marking this as `DNM` because this test is broken, which wasn't discovered since it wasn't tested with vstart_runner.py or teuthology. A fix has been written for it (#56846). Both of them needs to be backported together to avoid passing same bug to release branches.

Fixed it

lxbsz commented:
This PR is under test in [https://tracker.ceph.com/issues/67316](https://tracker.ceph.com/issues/67316).

rishabh-d-dave reviewed (changes requested):
QA run was unsuccessful. From https://pulpito.ceph.com/xiubli-2024-08-05_05:14:48-fs-wip-xiubli-testing-20240802.055215-squid-distro-default-smithi/7836646 -

```
fail    7836646     0:31:34     fs/functional/{begin/{0-install 1-ceph 2-logrotate 3-modules} clusters/1a3s-mds-4c-client conf/{client mds mgr mon osd} distro/{centos_latest} mount/kclient/{mount-syntax/{v2} mount overrides/{distro/stock/{centos_9.stream k-stock} ms-die-on-skipped}} objectstore/bluestore-ec-root overrides/{ignorelist_health ignorelist_wrongly_marked_down no_client_pidfile pg_health} subvol_versions/create_subvol_version_v2 tasks/admin}    2
Failure Reason:

Test failure: test_multifs_single_path_rootsquash (tasks.cephfs.test_admin.TestFsAuthorize), test_multifs_single_path_rootsquash (tasks.cephfs.test_admin.TestFsAuthorize)
```

vshankar commented:
> QA run was unsuccessful. From https://pulpito.ceph.com/xiubli-2024-08-05_05:14:48-fs-wip-xiubli-testing-20240802.055215-squid-distro-default-smithi/7836646 -
> 
> ```
> fail    7836646     0:31:34     fs/functional/{begin/{0-install 1-ceph 2-logrotate 3-modules} clusters/1a3s-mds-4c-client conf/{client mds mgr mon osd} distro/{centos_latest} mount/kclient/{mount-syntax/{v2} mount overrides/{distro/stock/{centos_9.stream k-stock} ms-die-on-skipped}} objectstore/bluestore-ec-root overrides/{ignorelist_health ignorelist_wrongly_marked_down no_client_pidfile pg_health} subvol_versions/create_subvol_version_v2 tasks/admin}    2
> Failure Reason:
> 
> Test failure: test_multifs_single_path_rootsquash (tasks.cephfs.test_admin.TestFsAuthorize), test_multifs_single_path_rootsquash (tasks.cephfs.test_admin.TestFsAuthorize)
> ```

@kotreshhr PTAL.

vshankar commented:
> > QA run was unsuccessful. From https://pulpito.ceph.com/xiubli-2024-08-05_05:14:48-fs-wip-xiubli-testing-20240802.055215-squid-distro-default-smithi/7836646 -
> > ```
> > fail    7836646     0:31:34     fs/functional/{begin/{0-install 1-ceph 2-logrotate 3-modules} clusters/1a3s-mds-4c-client conf/{client mds mgr mon osd} distro/{centos_latest} mount/kclient/{mount-syntax/{v2} mount overrides/{distro/stock/{centos_9.stream k-stock} ms-die-on-skipped}} objectstore/bluestore-ec-root overrides/{ignorelist_health ignorelist_wrongly_marked_down no_client_pidfile pg_health} subvol_versions/create_subvol_version_v2 tasks/admin}    2
> > Failure Reason:
> > 
> > Test failure: test_multifs_single_path_rootsquash (tasks.cephfs.test_admin.TestFsAuthorize), test_multifs_single_path_rootsquash (tasks.cephfs.test_admin.TestFsAuthorize)
> > ```
> 
> @kotreshhr PTAL.

Gentle nudge on this @kotreshhr

mchangir commented:
This PR is under test in [https://tracker.ceph.com/issues/69882](https://tracker.ceph.com/issues/69882).

joscollin commented:
kotreshhr:wip-64939-squid is rebased by https://github.com/ceph/ceph/pull/65802 and it is under testing.
